resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/KubernetesDriverCustomFeatureConfigStep.scala

```diff
@@ -25,6 +25,45 @@ import org.apache.spark.deploy.k8s.KubernetesDriverConf
  * A base interface to help user extend custom feature step in driver side.
  * Note: If your custom feature step would be used only in driver or both in driver and executor,
  * please use this.
+ *
+ * Example of driver feature step:
+ *
+ * {{{
+ *   class DriverExampleFeatureStep extends KubernetesDriverCustomFeatureConfigStep {
+ *     private var driverConf: KubernetesDriverConf = _
+ *
+ *     override def init(conf: KubernetesDriverConf): Unit = {
+ *       driverConf = conf
+ *     }
+ *
+ *     // Implements methods of `KubernetesFeatureConfigStep`, such as `configurePod`
+ *     override def configurePod(pod: SparkPod): SparkPod = {
+ *       // Apply modifications on the given pod in accordance to this feature.
+ *     }
+ *   }
+ * }}}
+ *
+ * Example of feature step for both driver and executor:
+ *
+ * {{{
+ *   class ExecutorExampleFeatureStep extends KubernetesDriverCustomFeatureConfigStep
+ *       with KubernetesExecutorCustomFeatureConfigStep {
```

Comment:
Sorry for nit-picking but please check [indentation](https://github.com/databricks/scala-style-guide#spacing-and-indentation).

Reply:
ah, yes.
> For classes whose header doesn't fit in two lines, use 4 space indentation for its parameters, put each in each line, **put the extends on the next line with 2 space indent**, and add a blank line after class header.


This fix was required by https://github.com/apache/spark/pull/35496#discussion_r805153828 , but I think 2 space is right. will recover soon.